src/fabric/mod.rs

```diff
@@ -246,9 +231,17 @@ pub struct UniqueId {
 }
 
 #[derive(Clone, Debug, Copy)]
-pub enum Link {
-    Push { ideal: f32 },
-    Pull { ideal: f32 },
-    PullStiffness { ideal: f32, stiffness_factor: f32 },
-    Measure { length: f32 },
+pub struct Link {
+    ideal: f32,
+    material: usize,
```

Comment:
```suggestion
    material_id: usize,
```
Same suggestion

Reply:
I'd consider it an index and not an id (identifier) and I don't want to use material_index everywhere.